### PR TITLE
Upgrade k8s, knative dependencies and Golang 1.18

### LINKED ISSUE #3807: Upgrade to Go 1.18.x
When can we upgrade Go version to 1.18.x?  Some dependencies start to require Go 1.18 like #3806.

Upgrading to Go 1.18 gives us some great new language features such as generics.

### PR DESCRIPTION
**Pull request by astefanutti**
**Release Note**
```release-note
Upgrade Golang to 1.18
Upgrade k8s to v1.25.2
Upgrade controller-runtime to v0.13.1
Upgrade Knative to v1.8.3
Upgrade Kustomize to v4.5.4
```

Fixes #3807.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -60,7 +60,7 @@ jobs:
     - name: release-1.10.x nightly
       uses: ./.github/actions/release-nightly
       with:
-        goVersion: "1.17.x"
+        goVersion: "1.18.x"
```
Comment (squakez) [inline .github/workflows/release.yml:63]: We should not upgrade version in 1.10

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -81,7 +81,7 @@ jobs:
     - name: release-1.11.x nightly
       uses: ./.github/actions/release-nightly
       with:
-        goVersion: "1.17.x"
+        goVersion: "1.18.x"
```
Comment (squakez) [inline .github/workflows/release.yml:84]: We should not upgrade version in 1.11

**Review by squakez — COMMENTED** [review]
Watch out the nightly release as we use it for 3 different branches.

File: .github/workflows/release.yml (hunk 1 above)
Comment (astefanutti) [reply .github/workflows/release.yml]: Good catch! Updated.

File: .github/workflows/release.yml (hunk 2 above)
Comment (astefanutti) [reply .github/workflows/release.yml]: Done, thanks!

**Review by oscerd — APPROVED** [review]

**Review by squakez — APPROVED** [review]

**Pull request merged**